Vikta-API-testing-sample/src/test/java/com/test/api/category/CategoryTest.java

```diff
@@ -2,50 +2,63 @@
 
 import com.test.api.dto.CategoryDTO;
 import com.test.api.execution.BaseTest;
-import org.junit.jupiter.api.Test;
-
-import java.util.Arrays;
+import io.restassured.response.Response;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.Tag;
+import java.util.*;
 
 import static io.restassured.RestAssured.given;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.*;
+import static org.assertj.core.api.Assertions.assertThat;
 
-/**
- * Test for Category API
- */
 public class CategoryTest extends BaseTest {
 
-    /**
-     * Test ob category title verification
-     */
+    private int idCategoryToClean = 0;
+    private final String idQueryParam = "id";
+
+    @AfterEach
+    void cleanup(TestInfo testInfo) {
+        if (testInfo.getTags().contains("needs-cleanup")) {
+                given().spec(defaultRequestSpec())
+                        .queryParam(idQueryParam , idCategoryToClean)
+                        .when()
+                        .delete(getTestEnvironment().getCategoryPath());
+        }
+    }
+
+
     @Test
-    public void verifyGetCategory() {
+    public void canGetCategoryByCorrectId() {
         final int categoryId = 5;
         final String expectedTitle = "Street";
 
-        assertThat("Category title is unexpected",
+        assertThat("Category title should be %s, but is %s",
                 getExistingCategoryById(categoryId).getTitle(),
                 equalTo(expectedTitle));
     }
 
-    /**
-     * Test with wrong description given
-     */
     @Test
-    public void verifyGetCategoryWithUnexpectedDescription() {
+    public void canNotGetCategoryWithUnexpectedId() {
+        final  int categoryId = 0;
+
+        Response response = getResponseFromCategoryById(categoryId);
+
+        assertThat(String.format("Category id: %s is correct, excepted wrong" ,categoryId), response.getStatusCode(), equalTo(404));
+        assertThat("There shouldn't be such a category" + categoryId, response.getBody().asString(), containsString("No such Category entity"));
+    }
+
+    @Test
+    public void canNotGetCategoryWithUnexpectedDescription() {
```

Comment:
this test does not make a lot of sense. I know it was there at the start, but we should get rid of it. Basically, You're getting some category by existing id and asserting that its description is not equal to some made up String